src/asf_tools/raster.py

```diff
@@ -55,17 +55,20 @@ def read_as_masked_array(raster: str | Path, band: int = 1) -> np.ma.MaskedArray
         band: The raster band to read
 
     Returns:
-        data: The raster pixel data as a numpy MaskedArray
+        masked_raster: The raster pixel data as a numpy MaskedArray
     """
     log.debug(f'Reading raster values from {raster}')
     ds = gdal.Open(str(raster))
     raster_band = ds.GetRasterBand(band)
-    data = np.ma.masked_invalid(raster_band.ReadAsArray())
+    
     nodata = raster_band.GetNoDataValue()
-    if nodata is not None:
-        return np.ma.masked_values(data, nodata)
+    if nodata and not np.isnan(nodata):
+        masked_raster = np.ma.masked_values(raster_band, nodata)
+    else:
+        masked_raster = np.ma.masked_invalid(raster_band.ReadAsArray())
```

Comment:
Does this need to be in an else block? I think it would be fine to always mask invalids, and for case where nodata is some value and there is invalid data, we'd want to mask both, I think.

Reply:
> Does this need to be in an else block? I think it would be fine to always mask invalids, and for case where nodata is some value and there is invalid data, we'd want to mask both, I think.

@jhkennedy Good point. I also noticed that when I changed `if nodata is not None` to `if nodata`, it no longer caught 0.0, so I reverted that.